cmd/origin.go

```diff
@@ -48,6 +48,25 @@ var (
 		Short:        "Start the origin service",
 		RunE:         serveOrigin,
 		SilenceUsage: true,
+		PreRun: func(cmd *cobra.Command, args []string) {
+			// Checking these values has to happen here and not in init() because init
+			// doesn't have access to the actual values passed on the command line.
+			if ost := viper.GetString("Origin.StorageType"); ost == "s3" {
+				if !viper.IsSet("Origin.S3Region") || !viper.IsSet("Origin.S3ServiceUrl") {
+					cmd.PrintErrln("The --region, --service-url flags or equivalent config file entries are required when the origin is launched in S3 mode.")
+					os.Exit(1)
+				}
+			} else if ost == "posix" {
+				// We specifically DON'T want the region, service-url, and url-style flags if the mode is posix
+				if viper.IsSet("Origin.S3Region") || viper.IsSet("Origin.S3ServiceUrl") || viper.IsSet("Origin.S3UrlStyle") {
```

Comment:
Will having these set actually break the posix mode in anyway?

Reply:
No, it wouldn't break POSIX. But I think it's nice to be very visible to users when they're combining config options that don't make sense together.